src/karthik/FindFinalDirectionOFVehicle/findFinalDirectionOfVehicle.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { inputData } from './data';
+import {
+  findFinalDirectionOfVehicle,
+  validMoves
+} from './findFinalDirectionOfVehicle';
+
+describe('findFinalDirectionOfVehicle function', () => {
+  test('validity 1 for input data', () => {
+    expect(findFinalDirectionOfVehicle(inputData, 'NORTH')).toEqual('SOUTH');
+  });
+
+  test('validity 2 for expected directions', () => {
+    expect(findFinalDirectionOfVehicle(['STRAIGHT'], 'NORTH')).toEqual('NORTH');
+  });
+
+  test('validity 3 for expected directions', () => {
+    expect(findFinalDirectionOfVehicle(['RIGHT'], 'NORTH')).toEqual('EAST');
+  });
+
+  test('validity 4 for expected directions', () => {
+    expect(findFinalDirectionOfVehicle(['BACK'], 'NORTH')).toEqual('SOUTH');
+  });
+
+  test('validity 5 for expected directions', () => {
+    expect(findFinalDirectionOfVehicle(['LEFT'], 'NORTH')).toEqual('WEST');
+  });
```

Comment:
I think these tests doing same thing you can avoid unnecessary tests

Reply:
I have elaborated the purpose on the existing test cases. Please confirm if ok.